### PR TITLE
Add a new item to the WooCommerce Task List with a link to the MP wizard [MAILPOET-4814]

### PR DESCRIPTION
**Pull request by rodrigoprimo**
## Description

This PR adds a new item to the WooCommerce Task List with a link to the MP wizard (see ticket description for more details). It also adds a new Mixpanel event to track when the user arrives to the MP wizard via the WooCommerce task list item.

## Code review notes

- To check that Mixpanel tracking is working you can add `debug: true` to [/mailpoet/assets/js/lib/analytics.js#L9](https://github.com/mailpoet/mailpoet/blob/c368b130b84ca66dcee6b91bd99eb754de8cd330/mailpoet/assets/js/lib/analytics.js#L9) (see https://developer.mixpanel.com/docs/javascript#debug-mode).
- Since Mixpanel is not enabled before the user finishes the wizard, I had to store the information that we want to send an event to Mixpanel in a setting and use it later in a different request. I have some doubts about this implementation, and I left a few questions in inline comments.

## QA notes

- Go to /wp-admin/admin.php?page=wc-admin
- If the user never completed the MP Welcome Wizard, check that there is a link under `Things to do next` pointing the user to the MP wizard. If the user completed the wizard, the same link should point to the newsletters page. In the future, we will update this to point to the MP homepage (https://mailpoet.atlassian.net/browse/MAILPOET-4926).
- Check that the MP Welcome Wizard still works as expected.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-4814]

## After-merge notes

_N/A_

[MAILPOET-4814]: https://mailpoet.atlassian.net/browse/MAILPOET-4814?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/assets/js/src/newsletters/listings/standard.jsx
Comment (rodrigoprimo) [inline mailpoet/assets/js/src/newsletters/listings/standard.jsx]: I'm not sure if this is the best place to put the code that sends the Mixpanel event and updates the setting.

File: mailpoet/assets/js/src/newsletters/listings/standard.jsx
Comment (rodrigoprimo) [inline mailpoet/assets/js/src/newsletters/listings/standard.jsx]: Ideally, I would like to remove the setting instead of changing its value. But it seems there is no method in our API to do that and I was not sure if I should add one. Thoughts?

I also considered using WP options, but it seems there is no WP REST API endpoint for options.

File: mailpoet/assets/js/src/newsletters/listings/standard.jsx
Comment (rodrigoprimo) [inline mailpoet/assets/js/src/newsletters/listings/standard.jsx]: I opted not to refactor this file to TSX as I already spent way more time than I wanted on this ticket. I tried to do the refactor but ran into some issues that were not straightforward to fix and decided to leave it to another moment.

File: mailpoet/lib/WooCommerce/MailPoetTask.php
Comment (costasovo) [inline mailpoet/lib/WooCommerce/MailPoetTask.php]: It is just a detail, but could we use a return type hint instead of the PHP doc return parameter? Tyhehints are checked runtime PHPdoc might become outdated. Or is there some reason/risk why we can't use type hints?

File: mailpoet/assets/js/src/newsletters/listings/standard.jsx
Comment (costasovo) [reply mailpoet/assets/js/src/newsletters/listings/standard.jsx]: I see that [the SettingsController has a method for deleting a setting](https://github.com/mailpoet/mailpoet/blob/926620e8f82ca691f1c11f6e973a2f86a6117522/mailpoet/lib/Settings/SettingsController.php#L117) so we could add the endpoint.

File: mailpoet/assets/js/src/newsletters/listings/standard.jsx
Comment (costasovo) [reply mailpoet/assets/js/src/newsletters/listings/standard.jsx]: Hmm. That is tricky. I'm currently working on the new homepage so after the homepage lands Emails will no longer be the page where we send users from the wizard.

I think we might not to put it anywhere in React, but maybe a simple `document.addEventListener('DOMContentLoaded', trackWizardLoaded)` could work. It can be placed into a separate file just needs to be added to [webpack_admin_index.tsx](https://github.com/mailpoet/mailpoet/blob/671c7eed7db526acb21e78cb8b3e6ff381e31685/mailpoet/assets/js/src/webpack_admin_index.tsx) so that it is bundled within admin JS which we include on every MailPoet admin page.
The DOMContentLoaded event should be enough to make sure `MailPoet.trackEvent` is available. We don't need to wait for Mixpanel to load because this is handled inside the `MailPoet.trackEvent` ([#](https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/assets/js/src/analytics_event.js#L68-L75)).

What do you think about such an approach? Would that work for you?

**Review by costasovo — COMMENTED** [review]
Good stuff! Thank you for preparing a fix for the issue with incorrect doctypes in Woo.

I added one comment and tried to answer your questions.

File: mailpoet/lib/WooCommerce/MailPoetTask.php
Comment (rodrigoprimo) [reply mailpoet/lib/WooCommerce/MailPoetTask.php]: Good point. I used an example class from a WooCommerce extension and didn't pause to reflect on the fact that I could use return type hints instead of docblocks precisely for the reason that you mentioned. Thanks for pointing that out. It is now fixed.

File: mailpoet/assets/js/src/newsletters/listings/standard.jsx
Comment (rodrigoprimo) [reply mailpoet/assets/js/src/newsletters/listings/standard.jsx]: I considered this option, but I wasn't sure if that was a desirable endpoint or not. Since you also think it is a good approach, I added it in a new commit.

File: mailpoet/assets/js/src/newsletters/listings/standard.jsx
Comment (rodrigoprimo) [reply mailpoet/assets/js/src/newsletters/listings/standard.jsx]: I went with the approach that you suggested.

The only downside that I see is that we are adding this script to every single page. That is not a big deal since it is the admin and it applies only to MP pages. I considered adding a check to load it only in the home page, but I thought this check could be a bit brittle and break in the future due to changes that we might implement. So I used only the DOMContentLoaded event.

**Comment by rodrigoprimo** [comment]
Thanks for the review, @costasovo. I implemented the changes that you suggested. Could you please take another look at this PR?

File: mailpoet/assets/js/src/newsletters/listings/standard.jsx
Comment (costasovo) [reply mailpoet/assets/js/src/newsletters/listings/standard.jsx]: > The only downside that I see is that we are adding this script to every single page.

Good point! But in most cases, it only checks if `window.mailpoet_track_wizard_loaded_via_woocommerce` is set so I think it is not a big deal.

**Review by costasovo — DISMISSED** [review]
LGTM!

File: mailpoet/assets/js/src/newsletters/listings/standard.jsx
Comment (rodrigoprimo) [reply mailpoet/assets/js/src/newsletters/listings/standard.jsx]: Yeah, I agree

**Review by Aschepikov — DISMISSED** [review]
GJ!

**Comment by Aschepikov** [comment]
Hi, @rodrigoprimo there are some conflicts, can you resolve them pls

**Comment by rodrigoprimo** [comment]
@Aschepikov, thanks for the ping. I fixed the conflict. CircleCI is now failing but it doesn't seem to be related to this PR as trunk is also failing with the same error.

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**
